shared/ClrFacade/InternalRDotNetDataConverter.cs

```diff
@@ -574,13 +584,10 @@ private SymbolicExpression convertToList(Array array)
    {
       if (array.Rank > 1)
          throw new NotSupportedException("Generic array converter is limited to uni-dimensional arrays");
-
-      // CAUTION: The following, while efficient, means that more specialized converters
-      // will not be picked up.
-      var elementConverter = tryGetConverter(array.GetType().GetElementType());
```

Comment:
In the old way, we looked up the type converter once for an array. So, if it's an array of floats, we get a converter that converts floats. Simple types all have their own converter and then anything that's not simple is converted as an object. 

So it worked as long as the array had a specific enough type or was an array of all objects. We needed to change this so that specific types in the array could be converted by their own converters. If an array of objects has a `float`, we want to convert that element as a `real` and the rest as an `object`.

We saw this note ```CAUTION```, so we were wary.